pslogin/src/main/scala/WorldSessionActor.scala

```diff
@@ -4265,9 +4322,15 @@ class WorldSessionActor extends Actor with MDCContextAware {
       }
 
     case msg @ ChangeFireStateMessage_Start(item_guid) =>
-      log.info("ChangeFireState_Start: " + msg)
+      log.trace("ChangeFireState_Start: " + msg)
       if(shooting.isEmpty) {
         FindEquipment match {
+          //special case - suppress the decimator's alternate fire mode
+          case Some(tool : Tool)
```

Comment:
line 4329 & 4334, same case, is that normal ?

Reply:
It's not the same case.  This case is sibject to the guard conditionals on lines 4330 and 4331.

Reply:
I think @SouNourS means you have `case Some(tool : Tool)` twice?

The two could potentially be merged and if the guard condition for the phoenix is met then set `shooting = Some(item_guid)` but the rest can be shared?